frontend/src/components/NotificationSettingsModal/NotifcationSettingsModal.tsx

```diff
@@ -0,0 +1,209 @@
+  const [pushEnabled, setPushEnabled] = useState(true);
+  const [importantEnabled, setImportantEnabled] = useState(true);
+  const [generalEnabled, setGeneralEnabled] = useState(true);
+  const [connectRequestEnabled, setConnectRequestEnabled] = useState(true);
+  const [connectionAcceptedEnabled, setConnectionAcceptedEnabled] = useState(true);
+  const [connectionDeniedEnabled, setConnectionDeniedEnabled] = useState(true);
+  const [privateEventInvitationEnabled, setPrivateEventInvitationEnabled] = useState(true);
+  const [eventRsvpUpdatesEnabled, setEventRsvpUpdatesEnabled] = useState(true);
+  const [publicEventAnnouncementsEnabled, setPublicEventAnnouncementsEnabled] = useState(true);
+  const [networkEventAnnouncementsEnabled, setNetworkEventAnnouncementsEnabled] = useState(true);
+
+  return (
+    <section className='flex max-h-[70vh] w-full max-w-xl flex-col rounded-lg bg-white'>
+      {/*Header just has title and close  */}
+      <header className='flex flex-shrink-0 items-center justify-between border-b border-gray-200 p-6'>
+        <h2 className='text-xl font-semibold'>Notification Settings 🔔</h2>
+        <X
+          onClick={onClose}
+          className='h-6 w-6 cursor-pointer transition-transform duration-200 hover:text-red-500'
+        />
+      </header>
+
+      {/*Main section comprises all the notification toggles */}
+      <main className='flex-1 overflow-y-scroll px-8'>
+        {/*This is the Push Notifications toggle has its own little section */}
+        <div className='flex flex-col gap-2 border-b py-5'>
+          <div className='flex items-center justify-between'>
+            <span className='font-medium'> Push Notifications</span>
+            <Toggle enabled={pushEnabled} onToggle={() => setPushEnabled(!pushEnabled)} />
+          </div>
+          <p className='text-sm font-medium text-gray-500'>
+            Receive in-app banner notifications for important priority alerts only
+          </p>
+        </div>
+
+        {/*This section is for important notifications starts with toggle to disbale all or enable and you can select which ones */}
+        <div className='flex flex-col gap-6 border-b py-5'>
+          <div className='flex items-center justify-between'>
+            <span className='font-medium'> 🚨 Important Notifications</span>
+            <Toggle
+              enabled={importantEnabled}
+              onToggle={() => {
+                const newValue = !importantEnabled;
+                setImportantEnabled(newValue);
+                if (!newValue) {
+                  {
+                    /*if turning off important notifs turn off all children as well */
+                  }
+                  setConnectRequestEnabled(false);
+                  setConnectionAcceptedEnabled(false);
+                  setConnectionDeniedEnabled(false);
+                  setPrivateEventInvitationEnabled(false);
+                }
```

Comment:
For these complex toggles, let's move them to auxiliary functions like `handleImportantToggle`